File: 2023/UICPC Regional/Round #13/solutions/babybites/per.hs
solve:: [String] -> Int -> String
solve [] _ = "makes sense"
solve (h:t) i
   | h == "mumble" = solve t (i+1)
   | h == show(i)  = solve t (i+1)
   | otherwise     = "something is fishy"

parse :: String -> ([String], Int)
parse str = (words (lines str !! 1), 1)

main = interact (uncurry solve . parse)

File: 2023/UICPC Regional/Round #13/solutions/babybites/per.go
// I have no idea what I'm doing.

package main

import (
    "fmt"
    "strconv"
)

func main() {
    var n int

    fmt.Scanf("%d", &n)

    for i := 1; i <= n; i++ {
        var token string
        fmt.Scanf("%s", &token)
        if token != "mumble" && token != strconv.Itoa(i) {
            fmt.Printf("something is fishy");
            return;
        }
    }
    fmt.Printf("makes sense");
}
